tests/docker-swarm/main.yml

```diff
@@ -0,0 +1,181 @@
+---
+# vim: set ft=ansible:
+#
+# !!!NOTE!!! This playbook was tested using Ansible 2.1; it is recommended
+# that the same version is used.
+#
+# This playbook is a basic functional test for docker swarm
+#  - Initialize docker swarm
+#  - Create docker swarm service
+#  - Scale replicas up and down
+#  - Remove service
+#  - Leave swarm
+#
+
+- name: Docker Swarm - Setup
+  hosts: all
+  become: yes
+
+  tags:
+    - setup
+
+  vars_files:
+    - vars.yml
+
+  roles:
+    - role: docker_version_check
+      docker_major: 1
+      docker_minor: 12
+      tags:
+        - docker_version_check
+
+    # Subscribe if the system is RHEL
+    - role: redhat_subscription
+      when: ansible_distribution == 'RedHat'
+      tags:
+        - redhat_subscription
+
+  post_tasks:
+    # docker swarm will not work with live restore enabled
+    #  and must be turned off
+    - name: Determine if /etc/docker/daemon.json exists
+      stat:
+        path: /etc/docker/daemon.json
+      register: daemon_file
+
+    - name: Turn off live restore
+      replace:
+        dest: /etc/docker/daemon.json
+        regexp: 'true'
+        replace: 'false'
+      when: daemon_file.stat.exists
+
+    - name: Restart docker service
+      service:
+        name: docker
+        state: restarted
+      when: daemon_file.stat.exists
+
+- name: Docker Swarm - Basic Functional Test
+  hosts: all
+  become: yes
+
+  tags:
+    - basic
+
+  vars_files:
+    - vars.yml
+
+  tasks:
+    - name: Verify docker swarm is not running
+      command: docker info
+      register: di
+
+    - name: Fail if docker swarm is already running
+      fail:
+        msg: "Docker swarm is already running"
+      when:  "'Swarm: active' in di.stdout"
```

Comment:
I was looking for something similar to rpm-ostree status --json.  Maybe in the next few releases!